src/sparsezoo/objects/file.py

```diff
@@ -12,367 +12,304 @@
-        """
-        self.check_download()
+    def _validate_numpy(self, strict_mode):
+        if not load_numpy_list(self._path):
+            self._throw_error(
+                error_msg="Numpy file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
 
-        return self.path
+    def _validate_onnx(self, strict_mode):
+        if not onnx.load(self._path):
+            self._throw_error(
+                error_msg="Onnx file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
 
-    def check_download(
-        self,
-        overwrite: bool = False,
-        refresh_token: bool = False,
-        show_progress: bool = True,
-    ):
-        """
-        Check if the file has been downloaded, if not then call download()
+    def _validate_recipe(self, strict_mode):
+        # only validate whether a file is a recipe if sparseml is installed.
+        # this is optional, since we do not want to have an explicit dependency
+        # on sparseml in sparsezoo.
+        from sparseml.pytorch.optim import ScheduledModifierManager
 
-        :param overwrite: True to overwrite any previous file, False otherwise
-        :param refresh_token: True to refresh the auth token, False otherwise
-        :param show_progress: True to print tqdm progress, False otherwise
-        """
-        if not self.downloaded or overwrite:
-            self.download(overwrite, refresh_token, show_progress)
+        try:
+            manager = ScheduledModifierManager.from_yaml(self._path)  # noqa  F841
+        except Exception as error:  # noqa: F841
+            self._throw_error(
+                error_msg="Markdown file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
 
-    def download(
-        self,
-        overwrite: bool = False,
-        refresh_token: bool = False,
-        show_progress: bool = True,
-    ):
-        """
-        Downloads a sparsezoo file.
+    def _validate_model_card(self):
+        try:
+            with open(self._path, "r") as yaml_file:
+                yaml_str = yaml_file.read()
+
+            # extract YAML front matter from markdown recipe card
+            # adapted from
+            # https://github.com/jonbeebe/frontmatter/blob/master/frontmatter
+            yaml_delim = r"(?:---|\+\+\+)"
+            _yaml = r"(.*?)"
+            re_pattern = r"^\s*" + yaml_delim + _yaml + yaml_delim
+            regex = re.compile(re_pattern, re.S | re.M)
+            result = regex.search(yaml_str)
+            yaml_str = result.group(1)
+            yaml_dict = yaml.safe_load(yaml_str)
+            # returns a string "{domain}-{sub_domain}, if valid
+            # this makes the method reusable to fetch the integration
+            # name for the integration validation
+            return yaml_dict
+
+        except Exception as error:  # noqa: F841
+            logging.error(error)
```

Comment:
Absolutely! Thanks for catching this.

Reply:
It should not, otherwise we would never have the opportunity to check whether the .md file is a recipe (when we fail to confirm that it is a model card)

Reply:
yeah :D